### PR TITLE
Add new section in settings for Nextcloud apps

### PR DESCRIPTION
**Pull request by mpivchev**
This adds a new section in settings to promote our other apps:

- If the app is installed, it opens it without doing anything else. 

    - Note: Originally it was supposed to open the app and switch accounts (if account is logged in), but that is not supported by Talk or Notes yet. Separate issues have been made. See: https://github.com/nextcloud/ios/issues/2460. For now I suggest we do it this way and move this PR forward until we have the other issues done @tobiasKaminsky @jancborchardt 

- If the app is not installed, it leads to the app store page.
- If the "more apps" icon is clicked, it opens a web page with other Nextcloud apps.

Additionally, a new URL scheme was made to add support for other apps to open this app and switch accounts (if account is logged in):

```
nextcloud://open-and-switch-account?user=someUser&url=https://cloud.nextcloud.com
```

Finally, refactored some of the code and removed duplicate code.

### CODE REVIEW

File: iOSClient/More/NCMore.swift [hunk 1]
```diff
@@ -483,53 +498,8 @@ class NCMore: UIViewController, UITableViewDelegate, UITableViewDataSource {
...
-            frame.size.width = newWidth
-            frame.origin.x += space
-            super.frame = frame
-        }
-    }
-}
-
-class NCMoreUserCell: UITableViewCell {
-
-    @IBOutlet weak var displayName: UILabel!
```
Comment (mpivchev) [inline iOSClient/More/NCMore.swift:517]: These were separated in their own files

File: iOSClient/More/Cells/BaseNCMoreCell.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        selectedBackgroundView = selectionColor
+        backgroundColor = .secondarySystemGroupedBackground
+        layer.cornerRadius = defaultCornerRadius
+    }
```
Comment (mpivchev) [inline iOSClient/More/Cells/BaseNCMoreCell.swift:35]: This code was used for every cell, so I made a base class instead.

**Comment by tobiasKaminsky** [comment]
Can you use 
[image](https://github.com/nextcloud/desktop/blob/d8479fb1bafacbef2ecdb26516b9c7f513204f01/theme/white/more-apps.svg?plain=1)

for more icon?

**Comment by mpivchev** [comment]
> we could take the opportunity to migrate this simple view into swiftUI and the GUI does not convince me

Hmm, we could've decided to do this earlier before these changes were made. But I can try nonetheless.

> the GUI does not convince me

Any suggestions are welcome. We can maybe ask some of our UI designers to design a view.

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Nice! Design-wise, how about something standard like Apple also uses on the top of Contacts? That has proper spacing and all.

**Comment by mpivchev** [comment]
@jancborchardt you mean this part?

image

Since we only have 3 of these, should they be stacked left or be in the middle?

**Review by marcoambrosini — COMMENTED** [review]
I don't think that these shortcuts should be the first entry in the settings because they can be detrimental to usability and create friction. I would say that settings is probably the wrong place to advertise the presence of other apps, and I suggest something subtler and "dismissable". if we absolutely have to, what about moving them to the bottom of the list?
Also, would these be visible even for users that have these apps installed?

**Comment by mpivchev** [comment]
> I don't think that these shortcuts should be the first entry in the settings because they can be detrimental to usability and create friction. I would say that settings is probably the wrong place to advertise the presence of other apps, and I suggest something subtler and "dismissable". if we absolutely have to, what about moving them to the bottom of the list?

I see your point. I still need to move this forward however, so if @jancborchardt is ok with this I can move this section to the bottom.

**Comment by jancborchardt** [comment]
> Since we only have 3 of these, should they be stacked left or be centered?

@mpivchev is there any recommended default or standard? Centered seems good.

---

@marcoambrosini the issue if they are on the bottom is that they will be difficult to discover, defeating the purpose. It is obvious that the vertical list of entries goes on, but it would not be obvious that there is an entirely new element below.

**Comment by mpivchev** [comment]
@jancborchardt Here is the updated look:
image

**Comment by mpivchev** [comment]
@jancborchardt Here is the new look:

**Comment by jancborchardt** [comment]
In iOS Contacts, there is a bit more top and bottom padding inside the entries. Can you adjust that @mpivchev? :)

**Comment by mpivchev** [comment]
@jancborchardt Is this better?

**Comment by mpivchev** [comment]
@jancborchardt pinging again in case you missed it

**Review by jancborchardt — APPROVED** [review]
Latest screenshot looks good! :)

**Comment by hannesfritz** [comment]
iOS seems to stretch those buttons if less are available. I just found an example with 4 instead of 5 like in the screenshot above. But I did an additional mockup to show what it might look with only the three options. 
In my opinion it is calmer on the eye, better aligned and additionally larger tap targets.

So just food for thought :) 
Contacts 4 jumps
Nextcloud more

**Comment by jancborchardt** [comment]
Yeah, that seems very nice @hannesfritz!

@mpivchev could you adjust them like that? :)

**Comment by mpivchev** [comment]
Branch restored so I can reimplement this soon

**Pull request merged**